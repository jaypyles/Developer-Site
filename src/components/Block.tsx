import React from "react";

type BlockProps = {
  link: string;
  small_description: string;
};

const Block: React.FC<BlockProps> = ({ link, small_description }) => {
  return (
    <div className="block flex flex-col bg-testAccent rounded mb-[1em]">
      <a href={link} className="link text-[75%] !ml-[0.5em]">
        {link}
      </a>
      <p className="description text-[60%] ml-[0.5em] mb-[0.5em]">
        {small_description}
      </p>
    </div>
  );
};

export default Block;
