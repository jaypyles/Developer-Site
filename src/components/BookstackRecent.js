import React, { useState, useEffect } from "react";
import Constants from "../constants";

const BookstackRecent = () => {
  const [pageURL, setPageURL] = useState("");
  const url = `${Constants.DOMAIN}/api/bookstack/recent-page`;

  const fetchBookstack = async () => {
    try {
      const res = await fetch(url);
      if (res.ok) {
        const data = await res.json();
        setPageURL(JSON.parse(data));
      } else {
        console.error("Failed to fetch from API: ", res.status);
      }
    } catch (error) {
      console.error("Failed to fetch from API: ", error);
    }
  };

  useEffect(() => {
    fetchBookstack();
  }, []);

  return (
    <div className="bookstack-recent">
      {pageURL ? (
        <p>
          Most recent note: <a href={pageURL.url}>{pageURL.url}</a>
        </p>
      ) : (
        <></>
      )}
    </div>
  );
};

export default BookstackRecent;
