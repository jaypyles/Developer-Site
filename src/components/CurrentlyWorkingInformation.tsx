import React from "react";

interface Data {
  image: string;
  title: string;
  description: string;
}

interface CurrentlyWorkingInformationProps {
  data: Data;
}

const CurrentlyWorkingInformation: React.FC<
  CurrentlyWorkingInformationProps
> = ({ data }) => {
  const imageUrl = `/images/${data.image}`;

  return (
    <div className="workblock flex flex-row mr-[0.5em] p-1 bg-imageAccent emboss">
      {imageUrl && (
        <img
          className="image w-[4em] h-[4em] mr-[0.5em]"
          src={imageUrl}
          alt={data.title}
        />
      )}
      <div className="content mt-[0.25]">
        <p className="title text-[75%] mb-0">{data.title}</p>
        <p className="description text-[65%] mb-0 text-gray-800">
          {data.description}
        </p>
      </div>
    </div>
  );
};

export default CurrentlyWorkingInformation;
