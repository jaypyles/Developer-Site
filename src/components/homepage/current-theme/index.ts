export { CurrentTheme } from "./current-theme";
