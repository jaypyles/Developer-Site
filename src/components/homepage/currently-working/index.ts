export { CurrentlyWorkingInformation } from "./currently-working";
