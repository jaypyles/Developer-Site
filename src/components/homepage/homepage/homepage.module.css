.homepage {
  background-color: var(--grey-secondary);
  padding: 1rem;
  position: relative;

  height: 100%;

  @media screen and (min-width: 922px) {
    height: auto;
  }
}

.aboutContent {
  display: flex;
  flex-direction: column;
  gap: 1rem;

  font-size: 1rem;
}

.projects {
  display: flex;
  flex-direction: column;

  gap: 1rem;
}

.buttonsWrapper {
  display: flex;
  justify-content: center;
  align-items: center;

  gap: 1rem;

  @media screen and (max-width: 767px) {
    flex-direction: column;
  }
}

.buttons {
  display: flex;
  justify-content: center;
  gap: 1rem;
  flex-wrap: wrap;
}

.popups span:hover {
  color: var(--accent) !important;
}
