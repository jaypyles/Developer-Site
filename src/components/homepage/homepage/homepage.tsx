import React from "react";
import { work } from "../../../data/working";
import { CurrentlyWorkingInformation } from "..";
import { LinkWithDescription } from "../../links";
import { Contacts } from "../../nav/contacts/contacts";
import { PopupState } from "src/lib/types";
import { retroButtons } from "../../../data/retro-buttons";

import { clsx } from "clsx";
import classes from "./homepage.module.css";

interface HomepageProps {
  handleOpen: (popup: keyof PopupState) => void;
  popupState: PopupState;
}

export const Homepage = ({ handleOpen, popupState }: HomepageProps) => {
  return (
    <div className={clsx("emboss-no-top", classes.homepage)}>
      <Contacts />
      <div className="about">
        <h1
          id="intro-heading"
          className="page-title text-lg font-prompt mb-[0.5em] w-fit"
        >
          Hey, I&apos;m Jayden üë®‚Äçüíª
        </h1>
        <div className={classes.aboutContent}>
          <p>I am a Software Engineer, working in web development at:</p>
          <div className={classes.currentlyWorking}>
            {work.map((data, index) => (
              <CurrentlyWorkingInformation key={index} imageData={data} />
            ))}
          </div>
          <p>
            This site will be a continuous work-in-progress, and will most
            likely change often. It is a playground for ideas and a learning
            tool for me to try out different and/or new technologies. This site
            is hosted on my own hardware, with even a few apps publicly
            available:
          </p>
          <div className={classes.projects}>
            <LinkWithDescription
              link="https://spaceify.jaydenpyles.dev"
              smallDescription="A hackathon project I and 3 others built in 24 hours."
              github="https://github.com/Space-ify/frontend-in-space"
            />
            <LinkWithDescription
              link="https://notes.jaydenpyles.dev"
              smallDescription="Custom built note rendering webapp."
              github="https://github.com/jaypyles/very-simple-notes"
            />
          </div>
          <div className={classes.popups}>
            <p>
              As an engineer, I have used many different&nbsp;
              <span
                className={`underline cursor-pointer ${
                  popupState.programmingLanguages
                    ? "text-gray-500"
                    : "text-black"
                }`}
                onClick={() => handleOpen("programmingLanguages")}
              >
                programming languages
              </span>
              &nbsp;and technologies. But I also enjoy a lot of different&nbsp;
              <span
                className={`underline cursor-pointer ${
                  popupState.coolBox ? "text-gray-500" : "text-black"
                }`}
                onClick={() => handleOpen("coolBox")}
              >
                hobbies
              </span>
              &nbsp;outside of programming.
            </p>
          </div>
          <LinkWithDescription
            link="https://jaydenpyles.dev/resume"
            smallDescription="View my resume."
          />
          <div className={classes.buttonsWrapper}>
            <div className={classes.buttons}>
              {retroButtons.map((link, index) => (
                <img key={index} src={link} />
              ))}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
