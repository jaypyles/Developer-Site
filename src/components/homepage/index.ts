export * from "./currently-working";
export * from "./homepage";
export * from "./current-theme";
