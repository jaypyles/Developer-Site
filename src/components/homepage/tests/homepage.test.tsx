import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Homepage } from "src/components/homepage";

const mockedStates = {
  programmingLanguages: true,
  coolBox: false,
};

test("Render homepage properly.", () => {
  const result = render(
    <Homepage handleOpen={jest.fn()} popupState={mockedStates} />
  );

  const headingElement = result.container.querySelector("#intro-heading");
  expect(headingElement).toMatchSnapshot();
});
