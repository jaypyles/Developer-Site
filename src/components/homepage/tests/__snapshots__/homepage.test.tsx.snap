// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render homepage properly. 1`] = `
<h1
  class="page-title text-lg font-prompt mb-[0.5em] w-fit"
  id="intro-heading"
>
  Hey, I'm Jayden 👨‍💻
</h1>
`;
